floss/main.py

```diff
@@ -356,9 +369,13 @@ def is_supported_file_type(sample_file_path: Path):
     :return: True if file type is supported, False otherwise
     """
     with sample_file_path.open("rb") as f:
-        magic = f.read(2)
+        magic = f.read(4)
 
     if magic in SUPPORTED_FILE_MAGIC:
+        FileType.ELF = True
+        return True
```

Comment:
let's refactor this function to `get_file_type` and raise/handle an Exception if it's not supported

Reply:
Noted.

Reply:
Changed the function.
Also should i just read the file type in main and pass it as parameters?